consensus/safety-rules/src/persistent_safety_storage.rs

```diff
@@ -71,7 +71,16 @@ impl PersistentSafetyStorage {
         execution_private_key: Ed25519PrivateKey,
         waypoint: Waypoint,
     ) -> Result<()> {
-        internal_store.import_private_key(CONSENSUS_KEY, consensus_private_key)?;
+        let result = internal_store.import_private_key(CONSENSUS_KEY, consensus_private_key);
+        // Attempting to re-initialized existing storage. This can happen in environments like
+        // cluster test. Rather than be rigid here, leave it up to the developer to detect
+        // inconsistencies or why they did not reset storage between rounds. Do not repeat the
+        // checks again below, because it is just too strange to have a partially configured
+        // storage.
+        if let Err(libra_secure_storage::Error::KeyAlreadyExists(_)) = result {
+            return Ok(())
```

Comment:
nit: I wonder if it's worth logging that initialization failed because the data already exists? At least then we won't get completely silent failures (e.g., if someone assumes this method completed all the way to the end)